src/nextjournal/clerk/analyzer.clj

```diff
@@ -153,12 +168,8 @@
                     (ana-jvm/macroexpand-1 form env))
         analyzed (analyze-form {#'ana/macroexpand-1 mexpander} (rewrite-defcached form))
         nodes (ana-ast/nodes analyzed)
-        vars (into #{}
-                   (comp (filter (comp #{:def} :op))
-                         (keep :var)
-                         (map symbol))
-                   nodes)
-
+        {:keys [vars declared]} (get-vars+forward-declarations nodes)
+        vars- (set/difference vars declared)
```

Comment:
we don't consider forward declarations among vars for building graph dependencies